### ISSUE TITLE
authenticateOidcSilent fallback?

### ISSUE THREAD

**Issue by coolhome** [perarnborg/vuex-oidc#124]
Setup:
Redirect URI
Silent Renew URI
Automatic Silent SignIn = true (with redirect in same vue app unlike demo app)
Automatic Silent Renew = true

Problem: 
Invalid state when silent renew fails because prompt=none does not satisfy IdP login response. The library starts to check access, initiates a silent renew, loads previous user data from store. At this point I have an access token loaded, used in api calls. I have a user object from an invalid session. No automatic recovery like to do a popup or redirect signin. 

![image](https://user-images.githubusercontent.com/700740/91901282-00ab3780-ec6e-11ea-8c65-e9b644e715df.png)

I can't tell if this is a bug or by design. If its by design we should probably improve the documentation a little. I'm going to see what happens when I use the promise now being returned on silent renew. Maybe the proper solution is to subscribe to mutations like `setOidcError` and react.

![image](https://user-images.githubusercontent.com/700740/91901746-a9f22d80-ec6e-11ea-8075-d35ef9ddab18.png)

1 - You will notice that `_authenticateOidcSilent` will always execute the `then` on failure.
2 - We try to load the user and if we can't call `authenticate()` which triggers a redirect login. However in this case the users cookie expired with the IdP, and the IdP is requesting a login and the user is resolved from local storage.

**Comment by coolhome** [comment]
Pretty confident these are related

#19

**Comment by coolhome** [comment]
This snippet allowed me to workaround the issue, not as nicely as I would hope. I'm pretty positive we want to make `_authenticateOidcSilent` throw on error AND fallback to `authenticate` like it would if it could not find the user.


```js
    //- note: my namespace is 'oidc'
    store.subscribe((mutation, state) => {
      if(mutation.type == "oidc/setOidcError" && mutation.payload.context === "authenticateOidcSilent") {
        store.dispatch('oidc/authenticateOidc', {
          redirectPath: router.currentRoute.fullPath
        });
      }
    });
```

**Comment by perarnborg** [comment]
You are correct I beleive, the function should throw an error in this case! Will fix within short!

**Comment by apavelm** [comment]
Just wanted to report the same 👍

**Comment by perarnborg** [comment]
@coolhome Thanks for a detailed issue report! This should be fixed en v3.9.5.

**Comment by coolhome** [comment]
Worked perfectly! Thanks @perarnborg
